ksql-serde/src/main/java/io/confluent/ksql/serde/avro/AvroFormat.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.serde.avro;
+
+import com.google.common.collect.ImmutableSet;
+import io.confluent.kafka.schemaregistry.avro.AvroSchema;
+import io.confluent.ksql.serde.AbstractFormat;
+import io.confluent.ksql.serde.FormatInfo;
+import io.confluent.ksql.serde.KsqlSerdeFactory;
+import io.confluent.ksql.util.KsqlConstants;
+import java.util.Set;
+
+public final class AvroFormat extends AbstractFormat {
+
+  public static final String FULL_SCHEMA_NAME = "fullSchemaName";
```

Comment:
Humm... just looked at how `VALUE_AVRO_SCHEMA_FULL_NAME` in the WITH clause is wired up to this. (Same for DELIMITED).

Thinking...

1. Remember that we'll have key variants of these at some point soon.  I think what's there will work, i.e. we'll have to change `CreateSourceAsProperties.getFormatProperties()` to `getValueFormatProperties()` or something. Maybe if you plan ahead for key formats you may get a better long term design.  No biggie though.
2. `VALUE_AVRO_SCHEMA_FULL_NAME` should urgently be renamed to `VALUE_SCHEMA_FULL_NAME` is we're soon to have Json or Protobuf schema names in there.
3. `CreateSourceAsProperties` codifies the mapping from `VALUE_AVRO_SCHEMA_FULL_NAME` to `fullSchemaName`.  That works fine for our built in ones, but how would other formats expose their properties in the WITH clause?  Maybe the property names exposed by the formats should match the names used in the WITH clause, (with optional `VALUE_` or `KEY_` prefixes).

Reply:
1. good point, haven't thought this through much but IMO the Formats should be agnostic of key/value formats (if they are supposed to behave differently, then they should be two different formats)
2. yup, will definitely do that :) 
3. my intention going forward was to ditch this mapping from whats in the WITH clause to what is used internally and have them stick to what's in the with clause. I'll create a ticket describing this

Reply:
https://github.com/confluentinc/ksql/issues/4444